stackslib/src/chainstate/burn/operations/stack_stx.rs

```diff
@@ -169,33 +177,59 @@ impl StackStxOp {
     fn parse_data(data: &Vec<u8>) -> Option<ParsedData> {
         /*
             Wire format:
-            0      2  3                             19        20
-            |------|--|-----------------------------|---------|
-             magic  op         uSTX to lock (u128)     cycles (u8)
+            0      2  3                             19           20    21                 54
+            |------|--|-----------------------------|------------|-----|-------------------|
+            magic  op         uSTX to lock (u128)     cycles (u8) option     signer key
+                                                                  marker
 
              Note that `data` is missing the first 3 bytes -- the magic and op have been stripped
 
              The values ustx to lock and cycles are in big-endian order.
 
              parent-delta and parent-txoff will both be 0 if this block builds off of the genesis block.
+
+             "signer key" is encoded as follows: the first byte is an option marker
+               - if it is set to 1, the parse function attempts to parse the next 33 bytes as a StacksPublicKeyBuffer
+               - if it is set to 0, the value is interpreted as None
         */
 
-        if data.len() < 17 {
+        if data.len() < 18 {
```

Comment:
I think we either need to:

- add an epoch check here
- default to `0` if there is no data past the `cycle` (IMO this is better)

Otherwise, in epoch <2.5, this would be a consensus-breaking change